[Cocoa] Refactor logic for lookalike character sanitization when copying or sharing links

whsieh opened the pull request:
#### e0227fc7d5f7ad63bff6514bf21f77134eddeb64
<pre>
[Cocoa] Refactor logic for lookalike character sanitization when copying or sharing links
<a href="https://bugs.webkit.org/show_bug.cgi?id=249244">https://bugs.webkit.org/show_bug.cgi?id=249244</a>
rdar://103314642

Reviewed by Aditya Keerthi.

* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/PlatformMac.cmake:
* Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.h: Added.
* Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.mm: Added.
* Source/WebCore/PAL/pal/spi/cocoa/NetworkConnectionIntegritySPI.h: Added.

Add PAL softlinking and SPI headers for frameworks that are required for the &quot;network connection
integrity&quot; feature.

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::requestLookalikeCharacterStrings):

Add an IPC method to asynchronously fetch a set of lookalike character strings.

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/NetworkProcess/cocoa/NetworkConnectionIntegrityHelpers.h: Added.
* Source/WebKit/NetworkProcess/cocoa/NetworkConnectionIntegrityHelpers.mm: Added.

Add a helper file that contains the definition of `requestLookalikeCharacterStrings` (and any other
related helpers that are related to network connection integrity, which we&apos;ll require in subsequent
patches).

* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didCommitLoad):

`updateLookalikeCharacterStringsIfNeeded()` after committing the mainframe load.

(WebKit::WebPage::updateLookalikeCharacterStringsIfNeeded):

Refactor this code such that we request and cache lookalike string data in the networking process,
and then propagate these strings down into the web process.

* Source/WebKit/WebProcess/WebPage/WebPage.h:

Canonical link: <a href="https://commits.webkit.org/257879@main">https://commits.webkit.org/257879@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8a8eb6958c72f01fe90c13762442f8f9910a0d13

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100229 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9396 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33304 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109552 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169786 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104221 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10297 "Built successfully") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92646 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107441 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106005 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91067 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34471 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/22498 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77421 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3151 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23977 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3130 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9257 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43465 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5419 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4961 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
```diff
@@ -742,3 +742,7 @@
             (machtrap-number MSC_mach_msg2_trap))))
 #endif // HAVE(SANDBOX_MESSAGE_FILTERING)
+
+#if USE(APPLE_INTERNAL_SDK)
```
Does this need a `__has_include`? Same question for the other `sb.in`.

whsieh replied:
Added — thanks for the review!

whsieh committed:
[Cocoa] Refactor logic for lookalike character sanitization when copying or sharing links
https://bugs.webkit.org/show_bug.cgi?id=249244
rdar://103314642

Reviewed by Aditya Keerthi.

* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/PlatformMac.cmake:
* Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.h: Added.
* Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.mm: Added.
* Source/WebCore/PAL/pal/spi/cocoa/NetworkConnectionIntegritySPI.h: Added.

Add PAL softlinking and SPI headers for frameworks that are required for the "network connection
integrity" feature.

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::requestLookalikeCharacterStrings):

Add an IPC method to asynchronously fetch a set of lookalike character strings.

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/NetworkProcess/cocoa/NetworkConnectionIntegrityHelpers.h: Added.
* Source/WebKit/NetworkProcess/cocoa/NetworkConnectionIntegrityHelpers.mm: Added.

Add a helper file that contains the definition of `requestLookalikeCharacterStrings` (and any other
related helpers that are related to network connection integrity, which we'll require in subsequent
patches).

* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didCommitLoad):

`updateLookalikeCharacterStringsIfNeeded()` after committing the mainframe load.

(WebKit::WebPage::updateLookalikeCharacterStringsIfNeeded):

Refactor this code such that we request and cache lookalike string data in the networking process,
and then propagate these strings down into the web process.

* Source/WebKit/WebProcess/WebPage/WebPage.h:

Canonical link: https://commits.webkit.org/257879@main
